src/swarm-tui/Swarm/TUI/Controller/EventHandlers/Main.hs

```diff
@@ -0,0 +1,155 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Here is the main player configurable key event handler while playing the game.
+module Swarm.TUI.Controller.EventHandlers.Main (
+  mainEventHandlers,
+) where
+
+import Brick
+import Brick.Keybindings
+import Control.Lens as Lens
+import Control.Monad (unless, void, when)
+import Control.Monad.IO.Class (liftIO)
+import Swarm.Game.Scenario.Topography.Structure.Recognition (automatons)
+import Swarm.Game.Scenario.Topography.Structure.Recognition.Type (originalStructureDefinitions)
+import Swarm.Game.State
+import Swarm.Game.State.Substate
+import Swarm.Game.Step (finishGameTick)
+import Swarm.TUI.Controller.EventHandlers.Frame (runGameTickUI)
+import Swarm.TUI.Controller.UpdateUI (updateUI)
+import Swarm.TUI.Controller.Util
+import Swarm.TUI.Editor.Model (isWorldEditorEnabled, worldOverdraw)
+import Swarm.TUI.Model
+import Swarm.TUI.Model.Event (MainEvent (..), SwarmEvent (..))
+import Swarm.TUI.Model.Goal
+import Swarm.TUI.Model.UI
+import System.Clock (Clock (..), TimeSpec (..), getTime)
+
+-- | Main keybindings event handler while running the game itself.
+--
+-- See 'Swarm.TUI.Controller.handleMainEvent'.
+mainEventHandlers :: [KeyEventHandler SwarmEvent (EventM Name AppState)]
+mainEventHandlers = allHandlers Main $ \case
+  QuitEvent -> ("Open quit game dialog", toggleQuitGameDialog)
+  ViewHelpEvent -> ("View Help screen", toggleModal HelpModal)
+  ViewRobotsEvent -> ("View Robots screen", toggleModal RobotsModal)
+  ViewRecipesEvent -> ("View Recipes screen", toggleDiscoveryNotificationModal RecipesModal availableRecipes)
+  ViewCommandsEvent -> ("View Commands screen", toggleDiscoveryNotificationModal CommandsModal availableCommands)
+  ViewMessagesEvent -> ("View Messages screen", toggleMessagesModal)
+  ViewStructuresEvent -> ("View Structures screen", toggleDiscoveryModal StructuresModal (structureRecognition . automatons . originalStructureDefinitions))
+  ViewGoalEvent -> ("View scenario goal description", viewGoal)
+  HideRobotsEvent -> ("Hide robots for a few ticks", hideRobots)
+  ShowCESKDebugEvent -> ("Show active robot CESK machine debugging line", showCESKDebug)
+  PauseEvent -> ("Pause or unpause the game", whenRunning safeTogglePause)
+  RunSingleTickEvent -> ("Run game for a single tick", whenRunning runSingleTick)
+  IncreaseTpsEvent -> ("Increase game speed by one tick per second", whenRunning . modify $ adjustTPS (+))
```

Comment:
These should actually say "double game speed" and "halve game speed", not "increase/decrease by one tick per second".

Reply:
Ups, you are right. I saw the `+` and forgot it adds to log2 of speed.